week-12/homework/index.js

```diff
@@ -0,0 +1,131 @@
+let input = document.querySelector('.input');
+let addBtn = document.querySelector('.add-btn');
+let itemContainer = document.querySelector('.item-container');
+let deleteBtns = document.querySelectorAll('.delete-btn');
+let editBtns = document.querySelectorAll('.edit-btn');
+let currentTaskNum = 0;
+
+//Creates tasklist, takes taskNumber and input value
+let processTaskList = function (taskNum, inputValue) {
+    let newInput = document.createElement("input");
+    let deleteBtn = document.createElement("button");
+    let editBtn = document.createElement("button");
+    let saveBtn = document.createElement("button");
+    let newDiv = document.createElement("div");
+
+    deleteBtn.innerHTML = "X";
+    deleteBtn.setAttribute('delete-item-number', taskNum);
+    deleteBtn.classList.add('delete-btn');
+    deleteBtn.classList.add('btn');
+    deleteBtn.classList.add('btn-danger');
+    editBtn.innerHTML = "Edit";
+    editBtn.setAttribute('edit-item-number', taskNum);
+    editBtn.classList.add('edit-btn');
+    editBtn.classList.add('btn');
+    editBtn.classList.add('btn-primary');
+    saveBtn.innerHTML = "Save";
+    saveBtn.setAttribute('save-item-number', taskNum);
+    saveBtn.classList.add('save-btn');
+    saveBtn.classList.add('d-none');
+    saveBtn.classList.add('btn');
+    saveBtn.classList.add('btn-secondary');
+    newInput.type = "text";
+    newInput.setAttribute('disabled', 'true');
+    newInput.setAttribute('input-item-number', taskNum);
+    newInput.classList.add('input-item');
+    newInput.classList.add('form-control');
+    newInput.value = inputValue;
+    newDiv.classList.add("item");
+    newDiv.classList.add("d-flex");
+    newDiv.classList.add("pt-1")
+    newDiv.setAttribute('item-number', taskNum);
+
+    newDiv.appendChild(newInput);
+    newDiv.appendChild(deleteBtn);
+    newDiv.appendChild(editBtn);
+    newDiv.appendChild(saveBtn);
+
+    itemContainer.appendChild(newDiv);
+
+
+    input.value = '';
+
+    deleteBtns = document.querySelectorAll('.delete-btn');
```

Comment:
The logic below are not correct as each time you call `processTaskList`, you add MORE click event to all the delete buttons. 
What does it mean? `addEventListener` will add another new `event` (in this case `"click"`) to the element.

```
myButton.addEventListener('click', firstClick);
myButton.addEventListener('click', secondClick);
```

when you actually `click` the `myButton`, 2 functions will be executed -> `firstClick` then `secondClick`

So with the code below the `processsDeleteBtn` will be called many times depending on how many times you click the `Add` button